services/provider/server/server.go

```diff
@@ -1663,9 +1609,7 @@ func (s *OCSProviderServer) appendVolumeReplicationClassKubeResources(
 		return kubeResources, nil
 	}
 
-	//TODO: this is a ugly hack to generate the peerStorageID for generating replicationID
-	// peerStorageID should come from GetStorageClientInfo
-	peerStorageID, err := s.getPeerStorageID(ctx, storageCluster, consumer.Namespace, consumerConfig.GetRbdRadosNamespaceName())
+	peerStorageID, err := s.getPeerStorageID(consumer)
```

Comment:
Why do we need to call into a supporting function if we only do one unmarshal? you might need to use other information for different resources. I would say, unmarshel before calling any of the appends and have the entire struct available for use